cygnus-ngsi/doc/user_and_programmer_guide/adding_new_sink.md

```diff
@@ -1,149 +1,122 @@
 
-    public abstract class OrionSink extends AbstractSink implements Configurable {
-    
-    	Batch batch;
-    	
-		/**
-		 * Constructor
-		 */ 
-		public OrionSink() {
-			super();
-			...
-		} // OrionSink
-
-		@Override
-		public Status process() throws EventDeliveryException {
-			Channel ch = getChannel();
-			Transaction txn = ch.getTransaction();
-			txn.begin();
-			
-			for (int i = 0; i < batchSize; i++) {
-				Event event = ch.take();
-				NotifyContextRequest notification = parseEventBody(event);
-				accumulateInBatch(event.getHeaders(), notification);
-			} // for
-			
-			persistBatch(batch);
-			txn.commit();
-			return Status.READY;
-		} // process
-
-    	/**
-     	 * This is the method the classes extending this class must implement when dealing with a batch of events to be
-     	 * persisted.
-     	 * @param batch
-     	 * @throws Exception
-        */
-    	abstract void persistBatch(Batch batch) throws Exception;
-    
-    } // OrionSink   
+    fiware-cygnus/cygnus-common/src/main/java/com/telefonica/iot/cygnus/errors/
 
-The `process` method is responsible for getting the channel, initiating a Flume transaction, taking as many events from the channel as necessary to build a `Batch` object and processing it by calling the `persistBatch` method. Such a `persistBatch` method is the only piece of code a developer must create according to the logic of his/her sink.
+Once finished, accumulation results in a `NGSIBatch` object, which internally holds sub-batches per each notified/grouped destination (`notified-destinations` and `grouped-destinations` headers in the Flume event objects are inspected to created the sub-batches, depending on the configured `enable_grouping` value). Information within this `NGSIBatch` object is the one the specific implementation of the new sink must persist.
 
-Please notice that the `process` method handles all the possible errors that may occur during a Flume transaction by catching exceptions, especially those thrown by the abstract `persistBatch` method. There exists a collection of Cygnus-related exceptions whose usage is mandatory located at:
+Specific persistence logic is implemented by overwritting the only abstract method within `NGSISink`, i.e. `persistBatch(NGSIBatch)`:
 
-    fiware-cygnus/src/main/java/com/telefonica/iot/cygnus/errors/
+    abstract void persistBatch(NGSIBatch) throws Exception;
 
 [Top](#top)
 
-###<a name="section3.2"></a>Sink configuration
-In addition to extending `AbstractSink`, `OrionSink` implements the `Configure` interface which allows for parameterizing the new sink from the general Cygnus configuration file.
+###<a name="section2.4"></a>Inherited counters
+Because `NGSISink` extends `CygnusSink` the following counters are alrady available for retrieving statistics of any sink extending `NGSISink`:
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/948/commits/9c967430d834ef018f1d7e70b5a2ac29614bfdf8